tests/unit/zhmcclient/test_group.py

```diff
@@ -0,0 +1,261 @@
+             None,
+             HTTPError({'http-status': 400, 'reason': 5})),
+            ({'description': 'fake description X'},  # props missing
+             None,
+             HTTPError({'http-status': 400, 'reason': 5})),
+            ({'description': 'fake description X',
+              'name': 'a'},
+             ['object-uri', 'name', 'description'],
+             None),
+        ]
+    )
+    def test_group_manager_create(self, input_props, exp_prop_names, exp_exc):
+        """Test GroupManager.create()."""
+
+        group_mgr = self.client.groups
+
+        if exp_exc is not None:
+
+            with pytest.raises(exp_exc.__class__) as exc_info:
+
+                # Execute the code to be tested
+                group_mgr.create(properties=input_props)
+
+            exc = exc_info.value
+            if isinstance(exp_exc, HTTPError):
+                assert exc.http_status == exp_exc.http_status
+                assert exc.reason == exp_exc.reason
+
+        else:
+
+            # Execute the code to be tested.
+            group = group_mgr.create(properties=input_props)
+
+            # Check the resource for consistency within itself
+            assert isinstance(group, Group)
+            group_name = group.name
+            exp_group_name = group.properties['name']
+            assert group_name == exp_group_name
+            group_uri = group.uri
+            exp_group_uri = group.properties['object-uri']
+            assert group_uri == exp_group_uri
+
+            # Check the properties against the expected names and values
+            for prop_name in exp_prop_names:
+                assert prop_name in group.properties
+                if prop_name in input_props:
+                    value = group.properties[prop_name]
+                    exp_value = input_props[prop_name]
+                    assert value == exp_value
+
+    def test_group_repr(self):
+        """Test Group.__repr__()."""
+
+        faked_group1 = self.add_group(name='a')
+        group_mgr = self.client.groups
+        group1 = group_mgr.find(name=faked_group1.name)
+
+        # Execute the code to be tested
+        repr_str = repr(group1)
+
+        repr_str = repr_str.replace('\n', '\\n')
+        # We check just the begin of the string:
+        assert re.match(r'^{classname}\s+at\s+0x{id:08x}\s+\(\\n.*'.
+                        format(classname=group1.__class__.__name__,
+                               id=id(group1)),
+                        repr_str)
+
+    @pytest.mark.parametrize(
+        "input_props, exp_exc", [
+            ({'name': 'a'},
+             None),
+            ({'name': 'b'},
+             None),
+        ]
+    )
+    def test_group_delete(self, input_props, exp_exc):
+        """Test Group.delete()."""
+
+        faked_group = self.add_group(name=input_props['name'])
+
+        group_mgr = self.client.groups
+        group = group_mgr.find(name=faked_group.name)
+
+        if exp_exc is not None:
+
```

Comment:
maybe that is personal choice, but I am surprised how many newlines you put in your code.